doc/design/OrionSTHSink.md

```diff
@@ -199,14 +215,55 @@ A configuration example could be:
     cygnusagent.sinks.sth-sink.db_prefix = cygnus_
     cygnusagent.sinks.sth-sink.collection_prefix = cygnus_
 
-## Use cases
+[Top](#top)
+
+##<a name="section3"></a>Use cases
 Use `OrionSTHSink` if you are looking for a Json-based document storage about aggregated data not growing so much in the mid-long term.
 
-## Contact
+[Top](#top)
+
+##<a name="section4"></a>Implementation details
+###<a name="section4.1"></a>`OrionSTHSink` class
+As any other NGSI-like sink, `OrionSTHSink` extends the base `OrionSink`. The methods that are extended are:
+
+    void persist(Map<String, String>, NotifyContextRequest) throws Exception;
+    
+The context data, already parsed by `OrionSink` in `NotifyContextRequest`, is iterated and persisted in the MongoDB backend by means of a `MongoBackend` instance. Header information from the `Map<String, String>` is used to complete the persitence process, such as the timestamp or the destination.
+    
+    public void start();
+
+`MongoBackend` is created. This must be done at the `start()` method and not in the constructor since the invoking sequence is `OrionSTHSink()` (contructor), `configure()` and `start()`.
+
+    public void configure(Context);
+    
+A complete configuration as the described above is read from the given `Context` instance.
+
+[Top](#top)
+
+###<a name="section4.2"></a>`MongoBackend` class
```

Comment:
Is this the same class documented for OrionMongoSink? What a pitty Markdown doesn't allow include (http://stackoverflow.com/questions/4779582/markdown-and-including-multiple-files)...